utils.go

```diff
@@ -16,20 +17,33 @@ const (
 	TagsHTTPHeaderPrefix = "Open-Tracing-Trace-Tags-"
 )
 
-// PropagateSpanInHeader encodes Span `ctx` to `h` as a series of
-// HTTP headers. Values are URL-escaped.
-func PropagateSpanInHeader(
+// InjectSpanInHeader encodes Span `sp` in `h` as a series of HTTP headers.
+// Values are URL-escaped.
+func InjectSpanInHeader(
 	sp Span,
 	h http.Header,
-	propagator SpanPropagator,
-) {
-	contextIDMap, tagsMap := propagator.PropagateSpanAsText(sp)
-	for headerSuffix, val := range contextIDMap {
+) error {
+	// First, look for a GoHTTPHeader injector (our preference).
+	injector := sp.Tracer().Injector(GoHTTPHeader)
+	if injector != nil {
+		return injector.InjectSpan(sp, h)
+	}
+
+	// Else, fall back on SplitText.
+	if injector = sp.Tracer().Injector(SplitText); injector == nil {
+		return errors.New("No suitable injector")
+	}
+	carrier := NewSplitTextCarrier()
+	if err := injector.InjectSpan(sp, carrier); err != nil {
+		return err
+	}
+	for headerSuffix, val := range carrier.TracerState {
 		h.Add(ContextIDHTTPHeaderPrefix+headerSuffix, url.QueryEscape(val))
```

Comment:
I think the method should take the prefixes as arguments, to really make it general purpose util method.